copy.go

```diff
@@ -266,11 +283,17 @@ func copyGraph(ctx context.Context, src content.ReadOnlyStorage, dst content.Sto
 }
 
 // doCopyNode copies a single content from the source CAS to the destination CAS.
-func doCopyNode(ctx context.Context, src content.ReadOnlyStorage, dst content.Storage, desc ocispec.Descriptor) error {
+func doCopyNode(ctx context.Context, src content.ReadOnlyStorage, dst content.Storage, desc ocispec.Descriptor, ch chan<- CopyUpdate) error {
 	rc, err := src.Fetch(ctx, desc)
 	if err != nil {
 		return err
 	}
+	if ch != nil {
+		rc = &progressReader{
+			c: ch,
+			r: rc,
+		}
+	}
```

Comment:
Wrapping a `ReadCloser` may have perf penalties as some types may implement `io.WriteTo` which can trigger an optimization by the built-in `io.Copy()` method.

Reply:
Are you saying that it is performance impact by wrapping the `Read()` call? Or that by not having the wrapper also  implement `WriterTo`, if the original `ReadCloser` does, then we lose the ability to `WriteTo`? I think the second of these.

What do you suggest? That we check for `WriterTo`, like `io.Copy()` does, and if it does, implement `WriteTo()`.

How would we then capture the updates?